network/src/noise_wrapper/handshake.rs

```diff
@@ -307,40 +373,67 @@ mod test {
         let server_private = x25519::PrivateKey::generate(&mut rng);
         let server_public = server_private.public_key();
 
-        let client = NoiseWrapper::new(client_private);
-        let server = NoiseWrapper::new(server_private);
+        let (client_auth, server_auth) = if is_mutual_auth {
+            let client_id = PeerId::random();
+            let client_keys = NetworkPublicKeys {
+                identity_public_key: client_public,
+            };
+            let server_id = PeerId::random();
+            let server_keys = NetworkPublicKeys {
+                identity_public_key: server_public,
+            };
+            let trusted_peers = Arc::new(RwLock::new(
+                vec![(client_id, client_keys), (server_id, server_keys)]
+                    .into_iter()
+                    .collect(),
+            ));
+            let client_auth = HandshakeAuthMode::mutual(trusted_peers.clone());
+            let server_auth = HandshakeAuthMode::mutual(trusted_peers);
+            (client_auth, server_auth)
+        } else {
+            (HandshakeAuthMode::ServerOnly, HandshakeAuthMode::ServerOnly)
+        };
```

Comment:
This is kind of my concern :/

ServerOnly, as you call it, most certainly has a set of trusted peers if it is the client.